crates/telio-wg/src/wg.rs

```diff
@@ -544,24 +592,76 @@ impl State {
             } else {
                 return false;
             };
-            if !try_send(PeerState::Connecting, peer.clone()).await {
+
+            if !try_send(PeerState::Connecting, None, peer.clone()).await {
                 return false;
             }
 
-            if peer.is_connected() && !try_send(PeerState::Connected, peer.clone()).await {
+            if peer.is_connected()
+                && !try_send(PeerState::Connected, Some(LinkState::Up), peer.clone()).await
+            {
                 return false;
             }
         }
 
+        let current_link_state = |peer| {
+            if matches!(self.no_link_detection, NoLinkDetection::Disabled) {
+                return None;
+            }
+
+            let new_rx = to.peers.get(peer).and_then(|p| p.rx_bytes);
+            let old_rx = self.last_rx_timestamp.get(peer).map(|ts| ts.1);
+
+            if new_rx != old_rx {
+                if new_rx.is_some() {
+                    Some(LinkState::Up)
+                } else {
+                    None
+                }
+            } else {
+                let configurable_rtt = match self.no_link_detection {
+                    NoLinkDetection::Disabled => DEFAULT_CONFIGURABLE_RTT,
```

Comment:
This branch is never executed because of `match!()` at the beginning of the closure. Can you merge botch matches and do the logic only when `NoLinkDetection` is `Enabled`?